src/Analysis/Ast/Impl/Types/PythonClassType.cs

```diff
@@ -196,7 +197,7 @@ internal PythonClassType(string name, Location location)
                 // During processing of builtins module some types may not be available yet.
                 // Specialization will attach proper base at the end.
                 Debug.Assert(!objectType.IsUnknown());
-                if (!bases.Any(b => objectType.Equals(b))) {
+                if (!bases.Any(b => b.Name == "object" && b.DeclaringModule.Name == @"builtins")) {
```

Comment:
Why previous comparison didn't work? We can have two instances of `objectType`?

Reply:
This probably can be reverted. The problem was when real analysis had one builtins module and restored code had another one.